src/main/java/org/isf/telemetry/daemon/TelemetryDaemon.java

```diff
@@ -0,0 +1,166 @@
+	private static final String FILE_PROPERTIES = "telemetry.properties";
+	private static final int DEFAULT_DELAY = 30; // seconds
+
+	// Singleton instance
+	private static volatile TelemetryDaemon instance;
+	private Thread telemetryThread;
+
+	private TelemetryManager telemetryManager;
+	private TelemetryUtils telemetryUtils;
+	private Telemetry settings;
+
+	private boolean running;
+	private int customDelay = DEFAULT_DELAY;
+	private int updateSettingsCounter;
+	private boolean reloadSettings;
+
+	/**
+	 * Starts the Telemetry Daemon
+	 */
+	public void start() {
+		telemetryThread = new Thread(this);
+		telemetryThread.start();
+		setRunning(true);
+		setReloadSettings(true);
+	}
+
+	private TelemetryDaemon() {
+		super(FILE_PROPERTIES);
+		LOGGER.info("Telemetry daemon started...");
+		customDelay = myGetProperty("telemetry.daemon.thread.loop.seconds", DEFAULT_DELAY);
+		LOGGER.info("Telemetry daemon loop set to {} seconds.", customDelay);
+		this.telemetryManager = Context.getApplicationContext().getBean(TelemetryManager.class);
+		this.telemetryUtils = Context.getApplicationContext().getBean(TelemetryUtils.class);
+		this.settings = telemetryManager.retrieveSettings();
+	}
+
+	// Singleton instance getter
+	public static TelemetryDaemon getTelemetryDaemon() {
+		if (instance == null) {
+			synchronized (TelemetryDaemon.class) {
+				if (instance == null) {
+					instance = new TelemetryDaemon();
+				}
+			}
+		}
+		return instance;
+	}
+
+	public void run() {
+		while (running) {
+			if (reloadSettings) {
+				this.settings = telemetryManager.retrieveSettings();
+				setReloadSettings(false);
+			}
+			LOGGER.info("Telemetry module running ({})...", updateSettingsCounter);
+			boolean isSendingMessageServiceActive = settings != null && settings.getActive() != null ? settings.getActive().booleanValue() : false;
+			if (!isSendingMessageServiceActive) {
+				LOGGER.info("No data selected.");
+			} else if (isSendingMessageServiceActive && isTimeToSendMessage()) {
+				try {
+					// GeneralData.getGeneralData();
+					this.telemetryUtils.sendTelemetryData(telemetryUtils.retrieveDataToSend(settings.getConsentMap()),
+									GeneralData.DEBUG);
+				} catch (RuntimeException | OHException e) {
+					LOGGER.error("Something strange happened.");
+					LOGGER.debug(e.getMessage(), e);
+					stop();
+				}
+			}
+			try {
+				Thread.sleep((long) customDelay * 1000);
+				updateSettingsCounter++;
+			} catch (InterruptedException e) {
+				LOGGER.error(e.getMessage(), e);
+				stop();
+			}
+		}
+	}
+
+	private boolean isTimeToSendMessage() {
+		boolean isTime = settings.getSentTimestamp() == null || !TimeTools.isSameDay(settings.getSentTimestamp(), LocalDateTime.now());
+		LOGGER.info("Telemetry module : already sent today");
+		return isTime;
+	}
+
+	/**
+	 * @param running the running to set
+	 */
+	private void setRunning(boolean running) {
+		this.running = running;
+	}
+
+	/**
+	 * @param running the running to set
```

Comment:
This looks wrong; the parm is `reload` not `running`

Reply:
Fixed thanks!